.github/workflows/test.yml

```diff
@@ -10,26 +10,37 @@ on:
 
 jobs:
   build:
+    name: Build and verify package
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: hynek/build-and-inspect-python-package@v2
+        id: baipp
+
+    outputs:
+      python-versions: ${{ steps.baipp.outputs.supported_python_classifiers_json_array }}
 
+  test:
+    name: Test package
+    needs: [build]
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        python-version:
-        - 3.9
-        - "3.10"
-        - "3.11"
-        - "3.12"
+        python-version: ${{ fromJson(needs.build.outputs.python-versions) }}
```

Comment:
Nice, I did not know this.

Reply:
Hm, strange, that it does not work. It looks exactly like the example in the docs.

Reply:
The example uses: `needs: build-package`.
Here we use `needs: [build]`. Is there a difference, when explicitly using a list in needs?

Reply:
Ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds